### PR TITLE
Warn on missing trailing slashes in targets

### PR DESCRIPTION
**Pull request by guybedford**
Implements https://github.com/domenic/import-maps/issues/93, along with warning handlers for the tests.

There were a surprising number of tests that failed the warning that needed changing. It can be illustrative to look through these.

### CODE REVIEW

File: reference-implementation/lib/parser.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ const { tryURLParse, hasFetchScheme, tryURLLikeSpecifierParse } = require('./uti
 // Tentative, so better to centralize so we can change in one place as necessary (including tests).
 exports.BUILT_IN_MODULE_PREFIX = '@std/';
 
-exports.parseFromString = (input, baseURL) => {
+exports.parseFromString = (input, baseURL, warn = console.warn.bind(console)) => {
```
[the remark this answers is not in the document]
Comment (guybedford) [reply reference-implementation/lib/parser.js]: I take it you mean the direct usage here right? Let me know if I've misunderstood.

[the remark this answers is not in the document]
Comment (domenic) [reply reference-implementation/lib/parser.js]: I mean, don't take `warn` as a parameter, just use the global `console.warn()`.

[the remark this answers is not in the document]
Comment (guybedford) [reply reference-implementation/lib/parser.js]: Ahh, the Jest API didn't seem to provide a way to hook into this so I just did it manually.

File: reference-implementation/__tests__/resolving.js [hunk 2]
```diff
@@ -153,8 +153,8 @@ describe('Mapped using the "imports" key only (no scopes)', () => {
         "/lib/no.mjs": null,
         "./dotrelative/no.mjs": [],
 
-        "/": "/lib/slash-only.mjs",
-        "./": "/lib/dotslash-only.mjs",
+        "/": "/lib/slash-only.mjs/",
+        "./": "/lib/dotslash-only.mjs/",
```
Comment (domenic) [inline reference-implementation/__tests__/resolving.js]: These are URL-like, so shouldn't have this requirement, I think. I will try to fix.

Comment (domenic) [reply reference-implementation/__tests__/resolving.js]: I guess it comes down to https://github.com/domenic/import-maps/issues/91 though, so, maybe I won't do anything...

**Pull request merged**